extensions/parquet/table/src/main/java/io/deephaven/parquet/table/S3ParquetInstructions.java

```diff
@@ -0,0 +1,102 @@
+package io.deephaven.parquet.table;
+
+import io.deephaven.annotations.BuildableStyle;
+import org.immutables.value.Value;
+
+/**
+ * This class provides instructions intended for reading and writing parquet files to AWS S3 instances.
+ */
+@Value.Immutable
+@BuildableStyle
+public abstract class S3ParquetInstructions {
+
+    private final static int DEFAULT_MAX_CONCURRENT_REQUESTS = 50;
+    private final static int DEFAULT_READ_AHEAD_COUNT = 1;
+    private final static int DEFAULT_FRAGMENT_SIZE = 512 << 20; // 5 MB
+    private final static int MIN_FRAGMENT_SIZE = 8 << 10; // 8 KB
+    private final static int DEFAULT_MAX_CACHE_SIZE = 50;
+
+    public static Builder builder() {
+        return ImmutableS3ParquetInstructions.builder();
+    }
+
+    /**
+     * The AWS region name to use when reading or writing to S3.
+     */
+    public abstract String awsRegionName();
+
+    /**
+     * The maximum number of concurrent requests to make to S3.
+     */
+    @Value.Default
+    public int maxConcurrentRequests() {
+        return DEFAULT_MAX_CONCURRENT_REQUESTS;
+    }
+
+    /**
+     * The number of fragments to send asynchronous read requests for while reading the current fragment.
+     */
+    @Value.Default
+    public int readAheadCount() {
+        return DEFAULT_READ_AHEAD_COUNT;
+    }
+
+    /**
+     * The maximum size of each fragment to read from S3. The fetched fragment can be smaller than this in case fewer
+     * bytes remaining in the file.
+     */
+    @Value.Default
+    public int fragmentSize() {
+        return DEFAULT_FRAGMENT_SIZE;
+    }
+
+    /**
+     * The maximum number of fragments to cache in memory.
+     */
+    @Value.Default
+    public int maxCacheSize() {
```

Comment:
It's interesting that the max cache size being > read ahead count supports for callers to re-read older fragments (assuming the modulus math doesn't stomp over it). It would be interesting to do some benchmarking where read ahead count is fixed to whatever we think is the best, and then vary maxCacheSize.